### PR TITLE
HPA e2e tests: fixed waiting for service creation.

### LINKED ISSUE #32512: [k8s.io] [HPA] Horizontal pod autoscaling (scale resource: CPU) [k8s.io] ReplicationController light Should scale from 1 pod to 2 pods {Kubernetes e2e suite}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/28599/kubernetes-pull-build-test-e2e-gce/58426/

Failed: [k8s.io] [HPA] Horizontal pod autoscaling (scale resource: CPU) [k8s.io] ReplicationController light Should scale from 1 pod to 2 pods {Kubernetes e2e suite}

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/horizontal_pod_autoscaling.go:88
Expected error:
    <*errors.errorString | 0xc820c80740>: {
        s: "error while stopping RC: rc-light-ctrl: timed out waiting for \"rc-light-ctrl\" to be synced",
    }
    error while stopping RC: rc-light-ctrl: timed out waiting for "rc-light-ctrl" to be synced
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/autoscaling_utils.go:307
```

Happened on a presubmit run in #28599.

Previous issues for this test: #27443 #27835 #28900

### PR DESCRIPTION
**Pull request by jszczepkowski**
HPA e2e tests: fixed waiting for service creation. Fixes #32512.

### CODE REVIEW

**Comment by jszczepkowski** [comment]
@k8s-bot verify test this

File: test/e2e/autoscaling_utils.go [hunk 1]
```diff
@@ -396,7 +396,18 @@ func runServiceAndWorkloadForResourceConsumer(c clientset.Interface, internalCli
 	}
 	framework.ExpectNoError(framework.RunRC(controllerRcConfig))
 
-	// Make sure endpoints are propagated.
-	// TODO(piosz): replace sleep with endpoints watch.
-	time.Sleep(10 * time.Second)
+	// Wait for endpoints to propagate for the controller service.
```
Comment (piosz) [inline test/e2e/autoscaling_utils.go]: Did you check whether there is no similar function in test utils?

Comment (jszczepkowski) [inline test/e2e/autoscaling_utils.go]: There was! Fixed.

**Comment by jszczepkowski** [comment]
@piosz PTAL

**Comment by piosz** [comment]
LGTM

**Pull request merged**